export/client/clients/database-client.go

```diff
@@ -0,0 +1,132 @@
+ * @version: 0.5.0
+ *******************************************************************************/
+package clients
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/edgexfoundry/edgex-go/export"
+	"gopkg.in/mgo.v2/bson"
+)
+
+type DatabaseType int8 // Database type enum
+const (
+	INVALID DatabaseType = iota
+	MONGO
+	MOCK
+)
+
+const (
+	invalidStr = "invalid"
+	mongoStr   = "mongodb"
+	mockStr    = "mockdb"
+)
+
+// Add in order declared in Struct for string value
+var databaseArr = [...]string{invalidStr, mongoStr, mockStr}
+
+func (db DatabaseType) String() string {
+	if db >= INVALID && db <= MOCK {
+		return databaseArr[db]
+	}
+	return invalidStr
+}
+
+// Return enum value of the Database Type
+func GetDatabaseType(db string) DatabaseType {
+	if mongoStr == db {
+		return MONGO
+	} else if mockStr == db {
+		return MOCK
+	} else {
+		return INVALID
+	}
+}
+
+type DBClient interface {
+	// ********************** REGISTRATION FUNCTIONS *****************************
+	// Return all the registrations
+	// UnexpectedError - failed to retrieve registrations from the database
+	Registrations() ([]export.Registration, error)
+
+	// Add a new registration
+	// UnexpectedError - failed to add to database
+	AddRegistration(reg *export.Registration) (bson.ObjectId, error)
+
+	// Update a registration
+	// UnexpectedError - problem updating in database
+	// NotFound - no registration with the ID was found
+	UpdateRegistration(reg export.Registration) error
+
+	// Get a registration by ID
+	// UnexpectedError - problem getting in database
+	// NotFound - no registration with the ID was found
+	RegistrationById(id string) (export.Registration, error)
+
+	// Get a registration by name
+	// UnexpectedError - problem getting in database
+	// NotFound - no registration with the name was found
+	RegistrationByName(name string) (export.Registration, error)
+
+	// Delete a registration by ID
+	// UnexpectedError - problem getting in database
+	// NotFound - no registration with the ID was found
+	DeleteRegistrationById(id string) error
+
+	// Delete a registration by name
+	// UnexpectedError - problem getting in database
+	// NotFound - no registration with the ID was found
+	DeleteRegistrationByName(name string) error
+}
+
+type DBConfiguration struct {
+	DbType       DatabaseType
+	Host         string
+	Port         int
+	Timeout      int
+	DatabaseName string
+	Username     string
+	Password     string
+}
+
+var ErrNotFound error = errors.New("Item not found")
+var ErrUnsupportedDatabase error = errors.New("Unsuppored database type")
+var ErrInvalidObjectId error = errors.New("Invalid object ID")
+var ErrNotUnique error = errors.New("Resource already exists")
+
+// Return the dbClient interface
+func NewDBClient(config DBConfiguration) (DBClient, error) {
+	switch config.DbType {
+	case MONGO:
+		// Create the mongo client
+		mc, err := newMongoClient(config)
+		if err != nil {
+			fmt.Println("Error creating the mongo client: " + err.Error())
```

Comment:
Do not print to stdout. Maybe something like:
return nil, fmt.Error(""Error creating the mongo client: " + err.Error())